### PR TITLE
Add plugin_prefix option on main class

### PR DESCRIPTION
**Pull request by mmerfort**
This allows overriding the default which is set in params.pp and used in
plugin.pp.

* Add parameter to main class
* Add documentation

### CODE REVIEW

**Comment by ekohl** [comment]
Looks sane, but I wonder what your use case is. Do you rebuild all packages with a custom prefix?

**Comment by mmerfort** [comment]
The defined type foreman::plugin references foreman::plugin_prefix [here](https://github.com/theforeman/puppet-foreman/blob/master/manifests/plugin.pp#L3).

However this parameter does currently not exist.

**Comment by ekohl** [comment]
It should, since ```foreman``` inherits from ```foreman::params```.

**Comment by mmerfort** [comment]
That's true, however for Foreman Version 1.9 and 1.8 the README suggests setting plugin_prefix          to “ruby193-rubygem-foreman_” which is not easily possible without this change.

**Comment by ekohl** [comment]
That's a very good reason I should have come up with myself :)

**Comment by mmoll** [comment]
:+1: after fixing indentation in the header

**Comment by mmerfort** [comment]
I don't see the wrong indentation in my sources: https://github.com/SC-Networks/puppet-foreman/blob/plugin_prefix/manifests/init.pp

Any idea why they are shown this way in the commit?

**Comment by mmoll** [comment]
Are you, by any chance, using hard tabs instead of spaces?

File: manifests/init.pp [hunk 1]
```diff
@@ -19,6 +19,8 @@
 # $passenger_ruby_package::    	Package to install to provide Passenger libraries for the active Ruby
 #                              	interpreter
 #
+# $plugin_prefix::              String which is prepended to the package name
```
Comment (domcleal) [inline manifests/init.pp]: *plugin* package names

**Comment by mmoll** [comment]
As it turned out, it was a mix of tabs and spaces, I converted all tabs to spaces and also used "plugin package names". Danke @mmerfort!

**Comment by mmerfort** [comment]
Thank you very much.

**Pull request closed without merging**